### ISSUE TITLE
target-content() function

### ISSUE THREAD

**Issue by bertfrees** [braillespecs/braille-css#6]
Alternatively the function could be called ‘target-move()’ or ‘target-pull()’ or ‘content(&lt;url&gt;)’.

‘[target-move()](http://www.w3.org/TR/2007/WD-css3-gcpm-20070205/#the-target-move)’ and ‘[target-pull()](http://www.w3.org/TR/2010/WD-css3-gcpm-20100608/#footnotes)’ were removed from [CSS3GCPM]. ‘target-content()’ is a value for the ‘[content](http://www.princexml.com/doc/9.0/properties/content)’ property in [PRINCE].

The  function would get the contents of an element including structure, rather than only the text value which ‘target-text()’ returns. Presumably this would work like [move-to](http://www.w3.org/TR/2003/WD-css3-content-20030514/#move-to) (also deprecated): it doesn't change the DOM, and elements and pseudo-elements that have been moved inherit from their position in the DOM, not from their new position.

This function would be useful for flowing footnotes to the bottom of the page on which the footnote _reference_ occurs.

Related issues:
- [ ] https://github.com/snaekobbi/braille-css-spec/issues/3
- [ ] https://github.com/snaekobbi/pipeline-mod-braille/issues/12

**Comment by josteinaj** [comment]
@bertfrees `target-content()` is used for notes in NLBs stylesheet. I'm having some trouble styling the content that are pulled in with it. Then I saw this issue; is it not implemented yet? Maybe it currently just works the same as `target-text()`?

**Comment by josteinaj** [comment]
Sorry, seems to work. It's just a bit confusing where to put the styles (together with `content: target-content(attr($refattr));` or on a separate rule matching the element) so I have to experiment a bit.

**Comment by bertfrees** [comment]
It's like described above:
> elements and pseudo-elements that have been moved inherit from their position in the DOM, not from their new position

So if you have a note:

```xml
<aside epub:type="note" class="notebody" id="note001"> <p> this is a note </p> </aside>
```

...and you want to move it to the position of the noteref:

```xml
<a epub:type="noteref" class="noteref" id="noteref_1" href="#note001">*</a>
```

...like this:

```css
a[epub|type~=noteref]::alternate {
    display: block;
    content: target-content(attr(href));
}
```

...you need to put the styles of the newly generated block box in the above rule, but the styles of the nested p needs to be put in a `aside > p` rule.

**Comment by bertfrees** [comment]
Yes. `::before` and `::after` content is deliberately not copied.

> I'm unable to add that text conditionally based on whether or not there are `<p>` children in the note

Hmm, that will indeed not work. Your XSLT solution is probably the easiest way to solve this.
